Python/Product/PythonTools/PythonTools/Environments/VirtualEnv.cs

```diff
@@ -140,34 +159,37 @@ static class VirtualEnv {
                 }
             }
 
-            // Ensure the target directory exists.
-            Directory.CreateDirectory(dir);
-
-            using (var proc = ProcessOutput.Run(
-                factory.Configuration.InterpreterPath,
-                new[] { "-m", useVEnv ? "venv" : "virtualenv", name },
-                dir,
-                UnbufferedEnv,
-                false,
-                output
-            )) {
-                var exitCode = await proc;
-
-                if (output != null) {
-                    if (exitCode == 0) {
-                        output.WriteLine(Strings.VirtualEnvCreationSucceeded.FormatUI(path));
-                    } else {
-                        output.WriteLine(Strings.VirtualEnvCreationFailedExitCode.FormatUI(path, exitCode));
+            var workspaceFactoryProvider = provider.GetComponentModel().GetService<WorkspaceInterpreterFactoryProvider>();
+            using (workspaceFactoryProvider?.SuppressDiscoverFactories(forceDiscoveryOnDispose: true)) {
+                // Ensure the target directory exists.
+                Directory.CreateDirectory(dir);
+
+                using (var proc = ProcessOutput.Run(
+                    factory.Configuration.InterpreterPath,
+                    new[] { "-m", useVEnv ? "venv" : "virtualenv", name },
+                    dir,
+                    UnbufferedEnv,
+                    false,
+                    output
+                )) {
+                    var exitCode = await proc;
+
+                    if (output != null) {
+                        if (exitCode == 0) {
+                            output.WriteLine(Strings.VirtualEnvCreationSucceeded.FormatUI(path));
+                        } else {
+                            output.WriteLine(Strings.VirtualEnvCreationFailedExitCode.FormatUI(path, exitCode));
+                        }
+                        if (provider.GetPythonToolsService().GeneralOptions.ShowOutputWindowForVirtualEnvCreate) {
+                            output.ShowAndActivate();
+                        } else {
+                            output.Show();
+                        }
                     }
-                    if (provider.GetPythonToolsService().GeneralOptions.ShowOutputWindowForVirtualEnvCreate) {
-                        output.ShowAndActivate();
-                    } else {
-                        output.Show();
-                    }
-                }
 
-                if (exitCode != 0 || !Directory.Exists(path)) {
-                    throw new InvalidOperationException(Strings.VirtualEnvCreationFailed.FormatUI(path));
+                    if (exitCode != 0 || !Directory.Exists(path)) {
+                        throw new InvalidOperationException(Strings.VirtualEnvCreationFailed.FormatUI(path));
```

Comment:
Does it show something useful to the user, such as message box?

Reply:
Yes, in output window.